enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -147,7 +149,7 @@ async def start_kernel(self, *args, **kwargs):
         username = KernelSessionManager.get_kernel_username(**kwargs)
         self.log.debug("RemoteMappingKernelManager.start_kernel: {kernel_name}, kernel_username: {username}".
                        format(kernel_name=kwargs['kernel_name'], username=username))
-
+        self.connection_dir=os.path.join(os.path.expanduser('~'+username),".local/share/jupyter/runtime/")
```

Comment:
self.connection_dir is normally determined by JUPYTER_RUNTIME_DIR but if this is not in user Paths in which to search for connection files it breaks so I did another hacky update. https://github.com/jupyter/jupyter_client/blob/98faaf91d00b4fbf17b2fbf0b08b1fd23adc27bf/jupyter_client/connect.py#L191

Reply:
Would it be better to call [`jupyter_core.paths.get_home_dir()`](https://github.com/jupyter/jupyter_core/blob/46a9c512e9e461c848df4a2ea4d2b96c5e527804/jupyter_core/paths.py#L35) to, presumably get `/root`, then replace that portion of `jupyter_runtime_dir()` with the target user's home directory?  Otherwise, this isn't producing the correct path relative to the platform.

We should also ensure the user's home directory exists, since KERNEL_USERNAME is not necessarily reflective of a user.  (I suppose setuid will enforce that constraint though.)

How will these kinds of files be cleaned up?  

`self.connection_dir` is on the multi kernel manager which spans all kernel manager instances - so this isn't going to work.  I'm not sure the connection information needs to be isolated on a per user level.  It isn't today at least.